ForwardingDecision.cs

```diff
@@ -0,0 +1,43 @@
+/*
+Pax : tool support for prototyping packet processors
+Nik Sultana, Cambridge University Computer Lab, June 2016
+
+Use of this source code is governed by the Apache 2.0 license; see LICENSE.
+*/
+
+namespace Pax {
+
+  // A packet processor makes some sort of forwarding decision, in addition
+  // to possibly analysing/modifying the packet (and generating new packets).
+  // The forwarding decision makes clear whether the processor recommends to
+  // * drop the packet (i.e., nothing more is to be done with it)
+  // * forward to a single port
+  // * forward to multiple ports
+  //
+  // NOTE this idiom simulates using algebraic types in C#. It's values are
+  //      analysed using the Visitor pattern.
+  public abstract class ForwardingDecision {
+    private ForwardingDecision() {}
+
+    // FIXME can use singleton to avoid multiple allocations of this?
+    public sealed class Drop : ForwardingDecision {
+      public Drop () {}
+    }
+
+    public sealed class SinglePortForward : ForwardingDecision {
```

Comment:
If allocations are a concern, perhaps instances of SinglePortForward should be cached singletons since they are likely to be high volume?

Certainly the constructor could be private and accessed through a method, allowing for a change later without a change in interface.
E.g. `public SinglePortForward ForwardTo(int target_port) { return new SinglePortForward(target_port); }` or similar.

I'm not sure how a similar pattern could be applied to the `MultiPortForward` class - perhaps it is best left to the user to cache the instances they know to be frequently used (such as broadcast on all ports bar the ingress port, for each port).